src/docker-in-docker/install.sh

```diff
@@ -109,17 +109,85 @@ find_version_from_git_tags() {
+    local prefix=${3:-"tags/v"}
+    # Some repositories use "_" instead of "." for version number part separation, support that
+    local separator=${4:-"."}
+    # Some tools release versions that omit the last digit (e.g. go)
+    local last_part_optional=${5:-"false"}
+    # Some repositories may have tags that include a suffix (e.g. actions/node-versions)
+    local version_suffix_regex=$6
+    # Try one break fix version number less if we get a failure. Use "set +e" since "set -e" can cause failures in valid scenarios.
+    set +e
+        major="$(echo "${current_version}" | grep -oE '^[0-9]+' || echo '')"
+        minor="$(echo "${current_version}" | grep -oP '^[0-9]+\.\K[0-9]+' || echo '')"
+        breakfix="$(echo "${current_version}" | grep -oP '^[0-9]+\.[0-9]+\.\K[0-9]+' 2>/dev/null || echo '')"
+
+        if [ "${minor}" = "0" ] && [ "${breakfix}" = "0" ]; then
+            ((major=major-1))
+            declare -g ${variable_name}="${major}"
+            # Look for latest version from previous major release
+            find_version_from_git_tags "${variable_name}" "${repository}" "${prefix}" "${separator}" "${last_part_optional}"
+        # Handle situations like Go's odd version pattern where "0" releases omit the last part
+        elif [ "${breakfix}" = "" ] || [ "${breakfix}" = "0" ]; then
+            ((minor=minor-1))
+            declare -g ${variable_name}="${major}.${minor}"
+            # Look for latest version from previous minor release
+            find_version_from_git_tags "${variable_name}" "${repository}" "${prefix}" "${separator}" "${last_part_optional}"
+        else
+            ((breakfix=breakfix-1))
+            if [ "${breakfix}" = "0" ] && [ "${last_part_optional}" = "true" ]; then
+                declare -g ${variable_name}="${major}.${minor}"
+            else 
+                declare -g ${variable_name}="${major}.${minor}.${breakfix}"
+            fi
+        fi
+    set -e
+}
+
+
 # Function to fetch the version released prior to the latest version
 get_previous_version() {
     repo_url=$1
-    curl -s "$repo_url" | jq -r 'del(.[].assets) | .[0].tag_name'
+    variable_name=$2
+    err_msg=$3
+    # Fetch the response headers for the rate limit information and store them in a variable
+    headers=$(curl -s --head -H "Accept: application/json" "$repo_url")
+    # Extract the rate limit information from the headers
+    limit=$(echo "$headers" | awk '/x-ratelimit-limit/{print $2}')
+    remaining=$(echo "$headers" | awk '/x-ratelimit-remaining/{print $2}')
+    reset_epoch=$(echo "$headers" | awk '/x-ratelimit-reset/{print $2}')
+    reset_time=$(date -d@"$reset_epoch" +"%Y-%m-%d %H:%M:%S" 2>/dev/null)
+    # remove trailing \r from $remaining
+    remaining=$(echo "$remaining" | tr -d '[:space:]')
+    # convert remaining to an int value for comparison to be greater than or less than 0
+    remaining_int=$(printf "%d" "$remaining")
+    if [[ $remaining_int -gt 0 ]]; then
```

Comment:
```suggestion
    # Fetch the response headers for the rate limit information and store them in a variable
headers=$(curl -s --head -H "Accept: application/json" "$repo_url")

# Extract the rate limit information from the headers
read -r limit remaining reset_epoch <<<$(echo "$headers" | grep -oP 'x-ratelimit-(limit|remaining|reset): \K\d+')

reset_time=$(date -d@"$reset_epoch" +"%Y-%m-%d %H:%M:%S" 2>/dev/null)

# convert remaining to an int value for comparison to be greater than or less than 0
remaining_int=$(printf "%d" "$remaining")

if [[ $remaining_int -gt 0 ]]; then
```

Can you try if this simplified code works?

Reply:
thanks ! made changes !